### DISCUSSION TITLE
Updating ICriteria.SetFetchMode to ICriteria.Fetch

### DISCUSSION THREAD

**Discussion by ghost** [nhibernate/nhibernate-core#3385] [Q&A] [answered]
Hi, everybody!
Me and my team have updated a project that was using an old version of this package, we are currently struggling with the update because the method: 
`ICriteria.SetFetchMode(string associationPath, FetchMode mode)`
has been deprecated and we have to switch to use
`ICriteria.Fetch(SelectMode mode, string associationPath)`

The problem we have is that we can't find anywhere what are the **SelectMode** equivalents to FetchMode.Join and FetchMode.Select.
Can you please help me with this?

**Comment by bahusoid** [comment, marked as answer]
`FetchMode.Eager`,  `FetchMode.Join`  -> `SeletctMode.Fetch` 
`FetchMode.Lazy`,  `FetchMode.Select` -> `SelectMode.Skip`
`FetchMode.Default` -> `SelectMode.Undefined`
